lightning/src/routing/scoring.rs

```diff
@@ -678,25 +795,103 @@ impl<L: Deref<Target = u64>, T: Time, U: Deref<Target = T>> DirectedChannelLiqui
 			} else {
 				let negative_log10_times_2048 =
 					approx::negative_log10_times_2048(numerator, denominator);
-				self.combined_penalty_msat(amount_msat, negative_log10_times_2048, params)
+				Self::combined_penalty_msat(amount_msat, negative_log10_times_2048,
+					params.liquidity_penalty_multiplier_msat,
+					params.liquidity_penalty_amount_multiplier_msat)
+			}
+		};
+
+		if params.historical_liquidity_penalty_multiplier_msat != 0 ||
+		   params.historical_liquidity_penalty_amount_multiplier_msat != 0 {
+			// If historical penalties are enabled, calculate the penalty by walking the set of
+			// historical liquidity bucket (min, max) combinations (where the min is less than the
+			// max) and, for each, calculate the probability of success given our payment amount,
+			// then total the weighted average probability of success.
+			//
+			// We use a sliding scale to decide which value will be used for a given bucket - the
```

Comment:
I read this multiple time but still have a hard time parsing what exactly is meant by 'value' and 'receives' in this context. Could you elaborate this further/make it more explicit? Also, while below examples are given for alternative approaches we *don't* take, it seems that less time is spent here elaborating the approach that we *actually* follow?